tests/JustEat.StatsD.Tests/SocketTransportTests.cs

```diff
@@ -72,6 +73,17 @@ public static void SocketTransportIsNoopForEmptyArray()
             }
         }
 
+        [Fact]
+        public static void SocketTransportIsNoopForNullEndpoint()
+        {
+            var endpointSource = Mock.Of<IEndPointSource>();
+
+            using (var transport = new SocketTransport(endpointSource, SocketProtocol.IP))
+            {
+                transport.Send("teststat:1|c");
```

Comment:
So this test fails when your change is absent right?

Reply:
Yes, without my change it would throw a null reference exception when the sockets pool is being created.